sentry-android/src/main/java/io/sentry/android/AndroidSentryClientFactory.java

```diff
@@ -35,19 +39,60 @@
      * @param ctx Android Context.
      */
     public AndroidSentryClientFactory(Context ctx) {
-        this.ctx = ctx;
-
         Log.d(TAG, "Construction of Android Sentry.");
+
+        this.ctx = ctx.getApplicationContext();
     }
 
     @Override
     public SentryClient createSentryClient(Dsn dsn) {
+        if (!checkPermission(Manifest.permission.INTERNET)) {
+            Log.e(TAG, Manifest.permission.INTERNET + " is required to connect to the Sentry server,"
+                + " please add it to your AndroidManifest.xml");
+        }
+
+        Log.d(TAG, "Sentry init with ctx='" + ctx.toString() + "' and dsn='" + dsn + "'");
+
+        String protocol = dsn.getProtocol();
+        if (!(protocol.equalsIgnoreCase("http") || protocol.equalsIgnoreCase("https"))) {
+            throw new IllegalArgumentException("Only 'http' or 'https' connections are supported in"
+                + " Sentry Android, but received: " + protocol);
+        }
+
+        if ("false".equalsIgnoreCase(dsn.getOptions().get(DefaultSentryClientFactory.ASYNC_OPTION))) {
+            throw new IllegalArgumentException("Sentry Android cannot use synchronous connections, remove '"
+                + DefaultSentryClientFactory.ASYNC_OPTION + "=false' from your DSN.");
+        }
+
         SentryClient sentryClient = super.createSentryClient(dsn);
         sentryClient.addBuilderHelper(new AndroidEventBuilderHelper(ctx));
+        SentryUncaughtExceptionHandler.setup();
         return sentryClient;
     }
 
     @Override
+    public Dsn lookupDsn() {
+        String stringDsn = "";
+
+        // attempt to get DSN from AndroidManifest
+        ApplicationInfo appInfo = null;
+        try {
+            PackageManager packageManager = ctx.getPackageManager();
+            appInfo = packageManager.getApplicationInfo(ctx.getPackageName(), PackageManager.GET_META_DATA);
+            stringDsn = appInfo.metaData.getString("io.sentry.android.DSN");
+        } catch (PackageManager.NameNotFoundException e) {
+            // skip
+        }
+
+        if (TextUtils.isEmpty(stringDsn)) {
+            throw new NullPointerException("Sentry DSN is not set, you must provide it to"
+                + "Sentry.init or in your Android manifest file.");
+        }
+
+        return new Dsn(stringDsn);
+    }
```

Comment:
Also moved from `SentryAndroid.init`, the new `lookupDsn` method was added so that each factory could decide how to lookup a `Dsn`. It used to be hardcoded to `Dsn.dsnLookup` which couldn't take into account Android (or other) impls.